src/app/api/reminder/route.ts

```diff
@@ -0,0 +1,241 @@
+import { PrismaClient } from "@prisma/client";
+import { HttpStatusCode } from "axios";
+import { Resend } from "resend";
+import { EmailTemplate } from "../../components/emailTemplate/email-template";
+
+const prisma = new PrismaClient();
+const resend = new Resend(process.env.RESEND_API_KEY);
+
+// Utility function to safely extract error messages from unknown errors
+function getErrorMessage(error: unknown): string {
+  if (error instanceof Error) {
+    return error.message;
+  }
+  return String(error); // Fallback for non-Error objects
+}
+
+// Helper function to send an email via Resend
+async function sendEmail(
+  subject: string,
+  actionType: "created" | "updated" | "deleted",
+  templateData: any
+) {
+  try {
+    const { data, error } = await resend.emails.send({
+      from: "OrthoChat <onboarding@resend.dev>",
+      to: ["alian.haidar01@gmail.com"], // Hardcoded email recipient
+      subject: subject,
+      react: EmailTemplate({ ...templateData, actionType })
+    });
+
+    if (error) {
+      console.error("Failed to send email:", error);
+      throw new Error("Failed to send email");
+    }
+
+    return data;
+  } catch (err) {
+    console.error("Error sending email:", err);
+  }
+}
+
+// GET: Fetch reminders
+export async function GET(request: Request) {
+  try {
+    const url = new URL(request.url);
+    const setById = url.searchParams.get("setById");
+    const setForId = url.searchParams.get("setForId");
+    const isReminderActive =
+      url.searchParams.get("isReminderActive") === "true";
```

Comment:
Could be simplified by:

```typescript
import { NextRequest, NextResponse } from 'next/server';

export async function GET(
  request: NextRequest,
) {
  //...

  // get search params
  const searchParams = request.nextUrl.searchParams;
  const isActive = searchParams.get("isActive");

  //...
}

```

Refer to https://nextjs.org/docs/app/api-reference/functions/next-request#nexturl